presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -120,24 +138,42 @@ public static CachingHiveMetastore memoizeMetastore(ExtendedHiveMetastore delega
                 newDirectExecutorService(),
                 OptionalLong.empty(),
                 OptionalLong.empty(),
-                maximumSize);
+                maximumSize,
+                false,
+                ALL);
     }
 
-    private CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor, OptionalLong expiresAfterWriteMillis, OptionalLong refreshMills, long maximumSize)
+    private CachingHiveMetastore(
+            ExtendedHiveMetastore delegate,
+            ExecutorService executor,
+            OptionalLong expiresAfterWriteMillis,
+            OptionalLong refreshMills,
+            long maximumSize,
+            boolean isPartitionVersioningEnabled,
+            MetastoreCacheScope metastoreCacheScope)
     {
         this.delegate = requireNonNull(delegate, "delegate is null");
         requireNonNull(executor, "executor is null");
+        this.isPartitionVersioningEnabled = isPartitionVersioningEnabled;
+
+        OptionalLong genExpiresAfterWriteMillis = metastoreCacheScope == ALL ? expiresAfterWriteMillis : OptionalLong.of(0);
```

Comment:
It might be easier if we set all these caches in a switch statement based on cache scope. It might be easier to understand logically, but I acknowledge there would be a lot of redundancy.

Otherwise, we could use switch statement to set these values instead of a lot of ternary expressions.

Reply:
Added a switch statement to set the cache configs.